src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +956,738 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+
+                if (!parameters.TryGetValue(fieldName, out object? param) || param is null)
+                {
+                    throw new DataApiBuilderException(message: "Mutation Request should contain a valid item field",
+                                                      statusCode: HttpStatusCode.BadRequest,
+                                                      subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+                }
+
+                // For point multiple create operation, the "parameters" dictionary is a key pair of <"item", List<ObjectFieldNode>>.
+                // The value field got using the key "item" cannot be of any other type.
+                // Ideally, this condition should never be hit, because such cases should be caught by Hotchocolate but acts as a guard against using any other types with "item" field
+                if (param is not List<ObjectFieldNode> paramList)
+                {
+                    throw new DataApiBuilderException(message: "Unsupported type used with 'items' field in the create mutation input",
+                                                      statusCode: HttpStatusCode.BadRequest,
+                                                      subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+                }
+
+                PerformDbInsertOperation(context, paramList, sqlMetadataProvider, multipleCreateStructure, primaryKeysOfCreatedItem);
+
+                if (multipleCreateStructure.CurrentEntityPKs is not null)
+                {
+                    primaryKeysOfCreatedItemsInTopLevelEntity.Add(multipleCreateStructure.CurrentEntityPKs);
+                }
+            }
+
+            return primaryKeysOfCreatedItemsInTopLevelEntity;
+        }
+
+        /// <summary>
+        /// 1. Identifies the order of insertion into tables involved in the create mutation request.
+        /// 2. Builds and executes the necessary database queries to insert all the data into appropriate tables.
+        /// </summary>
+        /// <param name="context">Hotchocolate's context for the graphQL request.</param>
+        /// <param name="parameters">Mutation parameter arguments</param>
+        /// <param name="sqlMetadataProvider">SqlMetadataprovider for the given database type.</param>
+        /// <param name="multipleCreateStructure">Wrapper object for the current entity for performing the multiple create mutation operation</param>
+        /// <param name="primaryKeysOfCreatedItem">Dictionary containing the PKs of the created items.</param>
+
+        private void PerformDbInsertOperation(
+            IMiddlewareContext context,
+            object? parameters,
+            ISqlMetadataProvider sqlMetadataProvider,
+            MultipleCreateStructure multipleCreateStructure,
+            Dictionary<string, Dictionary<string, object?>> primaryKeysOfCreatedItem)
```

Comment:
Please correct me if I am wrong in understanding this but why does this need to be a dictionary of dictionary? As far as I understood, this needs to store the values of the fields for which the current entity received the values from its parent entity (i.e. current entity is the referencing entity). For every child, there is only one parent. So, we don't need a dictionary of dictionary, but rather just a dictionary should suffice. 

A question along these lines - Do we even need `HigherLevelEntityPKs`/`HigherLevelEntityName` to be a part of MultipleCreateStructure?

Reply:
No, we need this to be `Dictionary<string, Dictionary<string,object?>>`

`primaryKeysOfCreatedItem` are used to store the primary keys of items created in the same nesting level as the current entity. Here, it is possible that an entity is referencing two different entities through different fields. So, we need to be able to identify the PKs for each relationship separately in order to successfully populate the foreign keys
So, we need a `Dictionary<string, Dictionary<string, object?>>`.

But, the current entity can be referencing only one higher level entity ---> So, i) we need to pass the higher level entity's Pks ii) At the same time, it is sufficient to have a dictionary. Hence, `HigherLevelEntityPKs` is just a dictionary.


Explaining both of these through an example, 

Consider a mutation such as this:

```graphql
createbook(item: {
     title: "...",
     reviews: [
       {
          content: "...",
          websiteuser: {
               username: "..."
          }
         another_referencing_relationship{
              ...
          }
         
       },
       {
          content: "...",
          websiteuser: {
               username: "..."
          }
          another_referencing_relationship{
              ...
          }
       }
    ]
})

```

Here, `Review` references 3 entities - `Book`, `websiteuser`, `another_referencing_relationship`. 

When processing the `Book` entity (top-level entity), `Review` will get identified as a referencing relationship. As a result, first a `Book` item will be created in the database.

Following that, the flow will get into processing each `Review` item. Here, the PKs of the book item will be passed through the `HigherLevelEntityPKs` field. Another thing to note here, is that for all the elements in the `reviews` input array, the `HigherLevelEntityPKs` needs to remain the same.

When we processing a `Review` item, `websiteuser` and `another_referencing_relationship` will be identified as **referenced** entities, and the flow will proceed to create a `websiteuser` item and a `another_referencing_relationship` item. The PKs of both of these item will get populated in `primaryKeysOfCreatedItem`. 

When populating the relationship fields of `Review` item, PKs will be fetched from `primaryKeysOfCreatedItem` as well as `HigherLevelEntityPKs`. 

Hope, this makes it clear as to why we need both `primaryKeysOfCreatedItem` and `HigherLevelEntityPKs` as well as the data structure type choices for them.

Reply:
I've updated the logic to populate the relationship fields as soon as each `referencedRelationship` is processed. So, this way, there is no need to store all the relationship fields. We don't need the `primaryKeysOfCreatedItem` data structure anymore.